src/SPIN/fix_nve_spin.cpp

```diff
@@ -221,74 +227,74 @@ void FixNVESpin::init()
   }
 
   // set ptrs for fix precession/spin styles
-  
+
   // loop 1: obtain # of fix precession/spin styles
-  
+
   int iforce;
   for (iforce = 0; iforce < modify->nfix; iforce++) {
-    if (strstr(modify->fix[iforce]->style,"precession/spin")) {
+    if (utils::strmatch(modify->fix[iforce]->style,"^precession/spin")) {
       nprecspin++;
     }
   }
-  
+
   // init length of vector of ptrs to precession/spin styles
 
   if (nprecspin > 0) {
     lockprecessionspin = new FixPrecessionSpin*[nprecspin];
   }
-  
+
   // loop 2: fill vector with ptrs to precession/spin styles
 
   int count2 = 0;
   if (nprecspin > 0) {
     for (iforce = 0; iforce < modify->nfix; iforce++) {
-      if (strstr(modify->fix[iforce]->style,"precession/spin")) {
+      if (utils::strmatch(modify->fix[iforce]->style,"^precession/spin")) {
         precession_spin_flag = 1;
         lockprecessionspin[count2] = (FixPrecessionSpin *) modify->fix[iforce];
         count2++;
       }
     }
   }
-  
+
   if (count2 != nprecspin)
-    error->all(FLERR,"Incorrect number of fix precession/spin");
+    error->all(FLERR,"Incorrect number of precession/spin fixes");
 
   // set ptrs for fix langevin/spin styles
-  
+
   // loop 1: obtain # of fix langevin/spin styles
-  
+
   for (iforce = 0; iforce < modify->nfix; iforce++) {
-    if (strstr(modify->fix[iforce]->style,"langevin/spin")) {
+    if (utils::strmatch(modify->fix[iforce]->style,"^langevin/spin")) {
       nlangspin++;
     }
   }
-  
-  // init length of vector of ptrs to precession/spin styles
+
+  // init length of vector of ptrs to langevin/spin styles
 
   if (nlangspin > 0) {
-    locklangevinspin = new FixLangevinSpin*[nprecspin];
+    locklangevinspin = new FixLangevinSpin*[nlangspin];
```

Comment:
@julient31 ... and here we have a cut-n-paste bug that was detected by using static code analysis with coverity scan.